.github/workflows/replay-verify.yaml

```diff
@@ -0,0 +1,108 @@
+name: "replay-verify"
+on:
+  # Allow triggering manually
+  workflow_dispatch:
+    inputs:
+      GIT_SHA:
+        required: false
+        type: string
+        description: The git SHA1 to test. If not specified, it will use the latest commit on main.
+      NETWORK:
+        required: true
+        type: choice
+        options:
+          - testnet
+          - mainnet
+          - all
+        description: The network to replay. If not specified, it will use the default network.
+  schedule:
+    - cron: "0 9/12 * * *"
+  pull_request:
+    paths:
+      - ".github/workflows/replay-verify.yaml"
+  push:
+    branches:
+      - devnet
+      - testnet
+      - mainnet
+      - aptos-node-v*
+
+# cancel redundant builds
+concurrency:
+  # cancel redundant builds on PRs (only on PR, not on branches)
+  group: ${{ github.workflow }}-${{ (github.event_name == 'pull_request' && github.ref) || github.sha }}
+  cancel-in-progress: true
+
+jobs:
+  # This job determines the target of the replay-verify job. It's a separate job so that we can use the `needs` syntax to make the replay-verify job depend on it.
+  # The replay-verify job should run under these conditions:
+  # - If we're running on a PR, we want to run the replay-verify job on all networks. But with a shorter timeout so as to only test the functionality of the workflow
+  # - If we're running on a workflow_dispatch, we want to run the replay-verify job on the network specified in the workflow_dispatch.
+  # - If we're running on a schedule, we want to run both the testnet and mainnet replay-verify jobs. This is the primary way the replay-verify workflow will run
+  determine-networks:
+    runs-on: ubuntu-latest
+    outputs:
+      target-network: ${{ steps.determine-networks.outputs.target-network }}
+    steps:
+      - name: Determine target network
+        id: determine-networks
+        run: |
+          if [[ "${{ github.event_name }}" == "pull_request" ]]; then
```

Comment:
Lets put this in a script as well, bash is fine but unless theres a good reason to keep it in the action lets make the workflow more simple

Reply:
This logic is very tightly coupled to the workflow, so it makes sense imo to keep this within the workflow spec inline rather than separate it out (and potentially have it fall out of date)